backend/fms_core/template_importer/sheet_data.py

```diff
@@ -13,20 +13,24 @@
 
 
 class SheetData():
-    def __init__(self, name, dataframe, headers):
+    def __init__(self, name, dataframe, headers, shared_data=None):
         self.base_errors = []
         self.is_valid = None
         self.header_row_nb = None
         self.last_empty_row = None
         self.name = name
         self.dataframe = dataframe
         self.headers = headers
+        self.shared_data = shared_data # This is additional information that is not assigned to a specific row. None for xls templates.
 
-        for i, row_list in enumerate(self.dataframe.values.tolist()):
-            if row_list[:len(self.headers)] == self.headers:
-                self.dataframe.columns = row_list
-                self.header_row_nb = i
-                break
+        if self.shared_data is not None: # This is not defined for xls templates
+            self.header_row_nb = -1 # No header in dataframe for json files
```

Comment:
So far it's not clear to me the use of `shared_data`.

Reply:
This is to hold the information that would go above the header in the templates ... There is a few templates with values there already. In the case here we use it to tell appart json and xls templates.